java-client/src/test/java/co/elastic/clients/elasticsearch/ElasticsearchTestServer.java

```diff
@@ -117,11 +117,17 @@ public synchronized ElasticsearchTestServer start() {
             return this;
         }
 
-        Version version = Version.VERSION.major() < 8 ? new Version(7,17,5,false) : new Version(8,12,0,false);
+        // using latest snapshot version for generic tests
+        Version version = Version.VERSION.major() < 8 ? new Version(7, 17, 25, false) :
+            Version.parse(Version.VERSION.major() + "." + Version.VERSION.minor() + ".0-SNAPSHOT");
```

Comment:
Why `.0-SNAPSHOT` when we may already have `.0` (final), `.1`, etc?

Maybe `.0-SNAPSHOT` if `version.patch() == 0` and `version.patch() - 1` otherwise, which will also reduce downloads once `.0` is released?